### PR TITLE
Add flags to ignore device attachement errors

### PR DESCRIPTION
**Pull request by errordeveloper**
- `agent.ingnore_scsi_bus_scan_errors` to avoid #767
- `agent.ingnore_all_device_errors` to avoid any device errors  causing agent to die

### CODE REVIEW

File: device.go [hunk 1]
```diff
@@ -352,7 +352,10 @@ func scanSCSIBusImpl(scsiAddr string) error {
 // associated with the disk.
 func getSCSIDevPathImpl(s *sandbox, scsiAddr string) (string, error) {
 	if err := scanSCSIBus(scsiAddr); err != nil {
-		return "", err
+		if !ignoreSCSIBusScanErrs {
+			return "", err
+		}
+		agentLog.WithError(err).Warnf("SCSI bus scanl failed")
```
Comment (dagrh) [inline device.go:358]: typo 'scanl' rather than 'scan'

**Comment by devimc** [comment]
@errordeveloper thanks for raising this, however I'm not sure if we really want these new parameters, if there is an error adding a device or re-scannning a bus then we'll end up having an incomplete sandbox/container, isn't it?
Rootfs block devices hot added may not be detected (devicemapper case) because of the lack of SCSI support in the guest, did  you try it out?
IMO guest kernel must fully support SCSI whether it's intended to be used as kernel for a kata container

**Comment by errordeveloper** [comment]
@davimc the guest kernel I was testing does support SCSI from what I can tell by the [config](https://github.com/linuxkit/linuxkit/blob/87f2e7542484baf3a223aef7b3a61c7ba0f8cad7/kernel/config-4.19.x-x86_64#L1748). From my point of view these params are useful as means of debugging, this way I was actually able to use what I have here and confirm that my kernel of choice is functional. I just need to debug a few more thing if it doesn't work, but actually right now I see no issues here, could it be just that kata is trying to pass some host devices to the VM that I didn't even know existed? Or is a SCSI device being used to mount default service account secret into my pod?

**Comment by jodh-intel** [comment]
I agree with @devimc, we should not be ignoring errors. Obviously, it's fine to use this branch for your own private debug though.

A couple of thoughts:

- Have you compared your guest kernel config with the Kata guest kernel configs here: https://github.com/kata-containers/packaging/tree/master/kernel ?
- Have you enabled [full debug](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#enable-full-debug)?

**Comment by errordeveloper** [comment]
Let's have the debugging discussion in the original issue - #767.

**Comment by bergwolf** [comment]
Ditto, please do not add these code injections just for a special debugging case.
> agent.ingnore_all_device_errors to avoid any device errors causing agent to die

AFAICT, agent should not die in such case. It just logs the error and fail the corresponding grpc request. Did you actually see agent dying in your test? If so, we should fix it as it's certainly a bug.

**Comment by errordeveloper** [comment]
> Did you actually see agent dying in your test? If so, we should fix it as it's certainly a bug.

I don't know actually, it's possible that the agent does exit, but as a result VM gets thrown out, as it's failed cause as far as CRI is concerned...

**Comment by jodh-intel** [comment]
@errordeveloper - You could try running our data collection script that looks for interesting errors, etc as it might give us further information on the problem you are seeing:

- Run the following:
  ```sh
  $ sudo kata-collect-data.sh > /tmp/kata.log
  ```
- Review the contents of `/tmp/kata.log` (to make sure there are no private/sensitive details).
- If you are happy, pasting the contents of `/tmp/kata.log` as a comment here (or on #767).

**Comment by amshinde** [comment]
Closing this PR as this has gone stale. And we really dont want to add code injections as pointed out by @bergwolf .
@errordeveloper Please feel to reopen, once you have addressed the comments.

**Pull request closed without merging**